File: 3211-generate-binary-strings-without-adjacent-zeros/3211-generate-binary-strings-without-adjacent-zeros.py
class Solution:
    def validStrings(self, n: int) -> List[str]:
        res = ["0", "1"]
        if n == 1: return res
        for i in range(2,n+1):
            temp = []
            for j in range(len(res)):
                if res[j][-1] == '1':
                    temp.append(res[j] + "1")
                    temp.append(res[j] + "0")
                else:
                    temp.append(res[j] + "1")
            res = temp
        return res

File: 3211-generate-binary-strings-without-adjacent-zeros/3211-generate-binary-strings-without-adjacent-zeros.java
class Solution {

    private void backTrack(List<String> ans,StringBuilder sb,int n){
        if(sb.length()==n){
            ans.add(sb.toString());
            return;
        }

        sb.append('1');
        backTrack(ans,sb,n);
        sb.deleteCharAt(sb.length()-1); 


        // 0
        if(sb.length()==0 || sb.charAt(sb.length()-1) != '0'){
            sb.append('0');
            backTrack(ans,sb,n);
            sb.deleteCharAt(sb.length()-1); //backTrack
        }

    }

    public List<String> validStrings(int n) {
        List<String> ans = new ArrayList<>();
        backTrack(ans,new StringBuilder(),n);
        return ans;
    }
}